xgcm/grid.py

```diff
@@ -84,6 +111,14 @@ def __init__(self, ds, axis_name, periodic=True, default_shifts={}, coords=None)
         self._ds = ds
         self.name = axis_name
         self._periodic = periodic
+        if boundary not in _VALID_BOUNDARY:
+            raise ValueError(
+                f"Expected 'boundary' to be one of {_VALID_BOUNDARY}. Received {boundary!r} instead."
+            )
+        self.boundary = boundary
+        if fill_value is not None and not isinstance(fill_value, (int, float)):
+            raise ValueError(f"Expected 'fill_value' to be a number.")
+        self.fill_value = fill_value if fill_value is not None else 0.0
```

Comment:
Is this consistent with the old behavior? I honestly don't know, but I would want to preserve existing behavior. 
E.g. if you do `grid.interp(da, 'X', boundary='fill', fill_value=None)` did this result in filling with 0?